network-test/reward-transfer-check.js

```diff
@@ -0,0 +1,118 @@
+    for (let validator of validatorsArr) {
+        console.log("validator: " + validator);
+        // Master of Ceremony doesn't have payout key
+        if (validator === masterOfCeremony) {
+            console.log("masterOfCeremony ");
+            continue;
+        }
+        let result = await checkValidatorRewardTransfer(validator);
+        await sqlDao.addToRewardTransfer([new Date(Date.now()).toISOString(), (result.passed) ? 1 : 0, result.validator,
+            result.payoutKey, result.error, result.blockNumber, JSON.stringify(result.transferTx), JSON.stringify(result.otherTxs)]);
+    }
+    sqlDao.closeDb();
+}
+
+checkRewardTransfers();
+
+/**
+ * Checks reward transfer for the specified validator
+ * @param validator
+ * @returns {Promise.<{passed: boolean, validator: *, payoutKey: *, error: string, transferTx: {}, otherTxs: Array, blockNumber: string}>}
+ */
+async function checkValidatorRewardTransfer(validator) {
+    console.log('--- checkValidatorRewardTransfer(), validator: ' + validator);
+    let payoutKey = await KeysManagerContract.methods.getPayoutByMining(validator).call();
+    console.log('payoutKey: ' + payoutKey);
+    let result = {
+        passed: true,
+        validator: validator,
+        payoutKey: payoutKey,
+        error: "",
+        transferTx: {},
+        otherTxs: [],
+        blockNumber: ""
+    };
+    if (payoutKey === "0x0000000000000000000000000000000000000000") {
+        result.passed = false;
+        result.error = "Validator doesn't have payout key!";
+        console.log("Validator doesn't have payout key! ");
+        return result;
+    }
+
+    let currentTimestamp = Date.now();
+    let hourAgoTimestamp = currentTimestamp - 3600 * 1000;
+    let currentBlockNumber = (await web3.eth.getBlock("latest")).number;
+    let block;
+    // in case of errors in getting blocks
+    let maxAttempts = 1000;
+    for (let i = 0; i < maxAttempts; i++) {
+        try {
+            block = await web3.eth.getBlock(currentBlockNumber, true);
+            result.blockNumber = currentBlockNumber;
+            if (block && block.transactions) {
+                block.transactions.forEach(async function (tx) {
+                    if (validator === tx.from) {
+                        if (payoutKey === tx.to) {
+                            console.log("found transfer tx at block: " + currentBlockNumber + " hash: " + tx.hash + ", from: " + tx.from, +", to: " + tx.to);
+                            result.transferTx = tx;
+                        }
+                        else {
+                            console.log("other tx at block: " + currentBlockNumber + " hash: " + tx.hash + ", from: " + tx.from, +", to: " + tx.to);
+                            // check if payout key was just changed
+                            let newPayoutKey = await KeysManagerContract.methods.getPayoutByMining(validator).call();
+                            if (newPayoutKey !== tx.to) {
```

Comment:
can this ever happen? It seems they're always equal
https://github.com/poanetwork/poa-network-monitor/pull/6/files#diff-9aa422941b3dda97f565fe04268a8aceR82
https://github.com/poanetwork/poa-network-monitor/pull/6/files#diff-9aa422941b3dda97f565fe04268a8aceR44

Reply:
Normally this code will be executed only if payout key was changed after test started (as is described in the issue), getting all blocks from the last hour takes some time, then newPayoutKey and payoutKey will be different, newPayoutKey will be equal to the tx.to and test will pass. But it can happen probably very rare.
Checking (newPayoutKey !== tx.to) fails now on one validator on Sokol, all transactions are made to the other then payout key account. This seems like the key was changed but KeysManager returns the old one.